CUDADataFormats/BeamSpot/interface/BeamSpotCUDA.h

```diff
@@ -21,12 +20,22 @@ class BeamSpotCUDA {
   };
 
   BeamSpotCUDA() = default;
-  BeamSpotCUDA(Data const* data_h, cudaStream_t stream);
+  BeamSpotCUDA(cudaStream_t stream) { data_d_ = cms::cuda::make_device_unique<Data>(stream); }
```

Comment:
Just thinking out loud, an alternative would be to also allocate the data in the EDProducer and move in the `cms::cuda::device::unique_ptr`. Then the `ptr()` interface would not be needed.

I'm not convinced if that would be better though. E.g. I'd think that for a generic SoA the EDProducer should not have to deal with the memory allocation.

I suppose the main reason why allocation-in-EDProducer would look feasible here is the fact that `BeamSpotCUDA` is essentially a wrapper for `cms::cuda::device::unique_ptr`. Which leads me to another alternative of storing directly something along `cms::cuda::device::unique_ptr<BeamSpotPOD>` (assuming `BeamSpotCUDA::Data` would be renamed to, e.g., `BeamSpotPOD`).